### DISCUSSION TITLE
Kong on Kubernetes / after securing the admin api , how to instruct the manager GUI to use the appropriate authentication

### DISCUSSION THREAD

**Discussion by kdejaeger** [Kong/kubernetes-ingress-controller#5470] [Q&A]
So I installed kong for kubernetes, both the manager and admin api are publically exposed. I added basic auth on the manager gui. However, this manager gui seems to need 'just a url' reference to the admin api. Where is the authentication configuration part? Off course this admin api will need to be secured, else everyone can just mutate it. But is there any setting so that the manager web based gui can refer to the admin api in a secure way (f. ex. with a apikey or basic auth).

**Comment by programmer04** [comment]
Kubernetes YAMLs should be the single source of truth for Kong configuration on Kubernetes. Everything that is possible and makes sense should be doable with YAMLs. You don't need access to API or Kong Manager, so we don't expose it by default.

If you're looking for the UI, the supported solution is [Konnect](https://konghq.com/products/kong-konnect), you can read how to configure it with KIC [here](https://docs.konghq.com/konnect/gateway-manager/kic/#main).

> _If the above answers your question please click `Mark as answer` button to increase visibility for others._

**Comment by kdejaeger** [comment]
Indeed the admin api is not exposed by default. But the OSS manager GUI needs the env variable admin_gui_api_url to be an exposed url. So to use the OSS manager, one needs to expose the admin api. And so then you need to secure the admin api, since it needs to be exposed. And then my question is, how can the manager still connect to an admin api that is secured?
